File: Assets/AltTester/AltDriver/Commands/ObjectCommands/AltClickElement.cs
namespace Altom.AltDriver.Commands
{
    public class AltClickElement : AltCommandReturningAltElement
    {
        AltClickElementParams cmdParams;

        public AltClickElement(IDriverCommunication commHandler, AltObject altUnityObject, int count, float interval, bool wait) : base(commHandler)
        {
            cmdParams = new AltClickElementParams(
            altUnityObject,
             count,
             interval,
             wait);
        }
        public AltObject Execute()
        {
            CommHandler.Send(cmdParams);
            var element = ReceiveAltObject(cmdParams);

            if (cmdParams.wait)
            {
                var data = CommHandler.Recvall<string>(cmdParams);
                ValidateResponse("Finished", data);
            }
            return element;
        }
    }
}

File: Bindings~/java/src/main/java/ro/altom/alttester/Commands/ObjectCommand/AltClickElement.java
package ro.altom.alttester.Commands.ObjectCommand;

import ro.altom.alttester.IMessageHandler;
import ro.altom.alttester.AltObject;
import ro.altom.alttester.Commands.AltCommandReturningAltObjects;

public class AltClickElement extends AltCommandReturningAltObjects {
    /**
     * @param command The parameters
     */
    private AltTapClickElementParams params;

    /**
     * @param parameters
     * @param messageHandler
     */

    public AltClickElement(IMessageHandler messageHandler, AltTapClickElementParams parameters) {
        super(messageHandler);
        this.params = parameters;
        this.params.setCommandName("clickElement");
    }

    public AltObject Execute() {
        SendCommand(params);
        AltObject obj = ReceiveAltObject(params);

        if (params.getWait()) {
            String data = recvall(params, String.class);
            validateResponse("Finished", data);
        }

        return obj;
    }
}
